svtools/bedpetobed12.py

```diff
@@ -3,89 +3,186 @@
-                            str(bedpe.score),bedpe.o1,str(bedpe.s1),str(bedpe.e1),"204,204,204"]))
-            output_lines.append('\t'.join([bedpe.c2,str(bedpe.s2),str(bedpe.e2),''.join(["BND;ID=",bedpe.name,";AF=",bedpe.af,";STR=",bedpe.o1,bedpe.o2]),\
-                            str(bedpe.score),bedpe.o2,str(bedpe.s2),str(bedpe.e2),"204,204,204"]))
+                s1, e1 = self.distant_coordinates(bedpe.o1, bedpe.s1, bedpe.e1)
+                size_tuple1 = self.distant_block_sizes(bedpe.o1, bedpe.s1, bedpe.e1)
+                start_tuple1 = self.distant_block_starts(bedpe.o1, bedpe.s1, bedpe.e1)
+                output_lines.append(self.create_line(
+                    bedpe.c1, 
+                    s1, 
+                    e1, 
+                    name, 
+                    bedpe.score, 
+                    bedpe.o1, 
+                    color, 
+                    size_tuple1,
+                    start_tuple1))
+                s2, e2 = self.distant_coordinates(bedpe.o2, bedpe.s2, bedpe.e2)
+                size_tuple2 = self.distant_block_sizes(bedpe.o2, bedpe.s2, bedpe.e2)
+                start_tuple2 = self.distant_block_starts(bedpe.o2, bedpe.s2, bedpe.e2)
+                output_lines.append(self.create_line(
+                    bedpe.c2,
+                    s2,
+                    e2,
+                    name,
+                    bedpe.score,
+                    bedpe.o2,
+                    color,
+                    size_tuple2,
+                    start_tuple2))
         else:
-            output_lines.append('\t'.join([bedpe.c1,str(bedpe.s1),str(bedpe.e1),''.join(["BND;ID=",bedpe.name,";STR=",bedpe.o1,bedpe.o2]),\
-                            str(bedpe.score),bedpe.o1,str(bedpe.s1),str(bedpe.e1),"204,204,204"]))
-            output_lines.append('\t'.join([bedpe.c2,str(bedpe.s2),str(bedpe.e2),''.join(["BND;ID=",bedpe.name,";STR=",bedpe.o1,bedpe.o2]),\
-                            str(bedpe.score),bedpe.o2,str(bedpe.s2),str(bedpe.e2),"204,204,204"]))
-    if len(output_lines) > 0:
-        output_handle.write('\n'.join(output_lines) + '\n')
-            
+            name = self.bed12_name(bedpe.svtype, bedpe.name, bedpe.af, (bedpe.o1, bedpe.o2))
+            output_lines.append(self.create_line(
+                bedpe.c1,
+                bedpe.s1,
+                bedpe.e1,
+                name, 
+                bedpe.score,
+                bedpe.o1,
+                color))
+            output_lines.append(self.create_line(
+                bedpe.c2,
+                bedpe.s2,
+                bedpe.e2,
+                name, 
+                bedpe.score,
+                bedpe.o2,
+                color))
+        return output_lines
+
 def processBEDPE(bedpe_stream, name, dist, output_handle=sys.stdout):
     #Process the BEDPE file and convert each entry to SAM.
-    if name is not None or bedpe_stream == sys.stdin:
```

Comment:
I came to the conclusion that since we had a default name set at the command line that this was not longer relevant.